src/modules/UI/scenes/Plugins/EdgeProvider.tsx

```diff
@@ -398,19 +403,78 @@ export class EdgeProvider extends Bridgeable {
       return
     }
 
-    // Prepare the internal spend request:
-    const info: GuiMakeSpendInfo = {
-      currencyCode,
-      nativeAmount: result.nativeAmount,
-      publicAddress: result.legacyAddress || result.publicAddress,
+    if (currencyCode !== this.selectedCurrencyCode) {
+      throw new Error('URI currency code mismatch from chooseCurrencyWallet selected code')
+    }
+
+    // PUBLIC ADDRESS URI
+    let tokenId: string | undefined
+    if (this.selectedChainCode !== this.selectedCurrencyCode) {
+      tokenId = getTokenId(account, this.selectedWallet.currencyInfo.pluginId, currencyCode)
+    }
+    const spendInfo: EdgeSpendInfo = {
       customNetworkFee,
       metadata,
-      lockInputs,
-      uniqueIdentifier
+      spendTargets: [
+        {
+          // Prioritize legacyAddress first since the existence of a legacy address means that a legacy address
+          // was scanned. Plugins may translate a legacy address into a publicAddress and provide that as well
+          publicAddress: legacyAddress ?? publicAddress,
+          memo: uniqueIdentifier,
+          nativeAmount
+        }
+      ],
+      tokenId
     }
+    return this._requestSpendCommon({ lockInputs, orderId, spendInfo })
+  }
 
-    // Launch:
-    return this._makeSpendRequest(info, this.selectedWallet, orderId, this.selectedCurrencyCode)
+  /**
+   * Internal helper to launch the send confirmation scene.
+   */
+  async _requestSpendCommon({
+    lockInputs,
+    orderId,
+    spendInfo
+  }: {
+    lockInputs: boolean
+    orderId?: string
+    spendInfo: EdgeSpendInfo
+  }): Promise<EdgeTransaction | undefined> {
+    return new Promise((resolve, reject) => {
+      const lockTilesMap = lockInputs ? { address: true, amount: true, wallet: true } : undefined
+
+      this._navigation.push('send2', {
```

Comment:
We want to use `navigation.navigate` whenever possible. The `navigate` action "just works", since it figures out what to do based on the current context. Most of our navigation bugs come about because we used one of the other actions, like `navigation.push`, in a context where that may have been incorrect.

For instance, if a plugin scene gets mounted inside the tab navigator, the correct navigation may be a `jump` rather than a `push`. Using `navigate` will automatically figure this out for you.

Reply:
nice insight. will do, thanks